src/test/java/org/tarantool/ClientAsyncOperationsIT.java

```diff
@@ -120,21 +123,17 @@ void testAsyncError(AsyncOpsProvider provider) {
     @MethodSource("getAsyncOps")
     void testOperations(AsyncOpsProvider provider)
         throws ExecutionException, InterruptedException, TimeoutException {
-        TarantoolClientOps<Integer, List<?>, Object, Future<List<?>>> ops = provider.getAsyncOps();
+        TarantoolClientOps<List<?>, Object, Future<List<?>>> ops = provider.getAsyncOps();
 
         List<Future<List<?>>> futures = new ArrayList<>();
 
         futures.add(ops.insert(spaceId, Arrays.asList(10, "10")));
-        futures.add(ops.delete(spaceId, Collections.singletonList(10)));
-
-        futures.add(ops.insert(spaceId, Arrays.asList(10, "10")));
-        futures.add(ops.update(spaceId, Collections.singletonList(10), Arrays.asList("=", 1, "ten")));
-
         futures.add(ops.replace(spaceId, Arrays.asList(20, "20")));
+        futures.add(ops.insert(spaceId, Arrays.asList(30, "30")));
+
+        futures.add(ops.delete(spaceId, Collections.singletonList(10)));
```

Comment:
Starting with a schema caching, we cannot guarantee the order of the async requests. It may cause a test failing time by time (i.e. delete operation will be processed earlier than insert one).

Reply:
I would say that we cannot guarantee it even for 'schema unsafe' operations. (However this should be true for memtx operations: it is strance that you got fails here. Also we don't perform DDL operations here, so schema id should be constant. But I agree that it is good to leave in this test from assumption of an execution order that a certain database engine (memtx/vinyl) provides — it is not even guaranteed in a future. And maybe my assumption that this ordering rules is true for memtx is not actually works.)

I think we should wait to insert of [10, "10"] before thying to delete it. The same for other pairs of dependent operations. Or, better, fill a space with independent rows synchronously (say, using console) and perform each async operation on their own row. In this case we can verify both replace and upsert variants: when a row is exists or doesn't.

Reply:
@nicktorwald What about the idea to pre-fill needed rows with a console and perform each async op on its own row to make them completely independent?

Reply:
The same about testFireAndForgetOperations() and testFireAndForgetStringOperations().